### PR TITLE
Fix rendering of line direction symbols in labels, cleanup label rendering code a bit

### LINKED ISSUE #59107: Labels: Line direction symbols not displayed
### What is the bug or the crash?

I'm working on a railway project which features a multi-line layer containing the gradient and length of the track in per mille as a feature. The line is symbolized as dashed line with points on changes of inclination. When using QGIS 3.38 the line direction symbols (<, >) are not being displayed anymore. It worked with QGIS 3.34 LTR.

My current alternative solution is creating an additional symbology as a marker line with an arrow showing the line direction (so QGIS reads and processes the line direction). Now the labels overlap the arrow symbols.

Screenshot 2024-10-16 160443

### Steps to reproduce the issue

Use a multine-line layer with features that can be used as label. Enable line direction symbols and see that nothing happens. Worked in QGIS 3.34. LTR.

### Versions

QGIS version
3.38.3-Grenoble
QGIS code revision
37f9e6efee
Qt version
5.15.13
Python version
3.12.6
GDAL/OGR version
3.9.2
PROJ version
9.4.0
EPSG Registry database version
v11.004 (2024-02-24)
GEOS version
3.12.2-CAPI-1.18.2
SQLite version
3.45.1
PDAL version
2.6.3
PostgreSQL client version
16.2
SpatiaLite version
5.1.0
QWT version
6.2.0
QScintilla2 version
2.14.1
OS version
Windows 11 Version 24H2

Active Python plugins
GeoBasis_Loader
1.0
GeoCoding
2.19
db_manager
0.1.20
grassprovider
2.12.99
MetaSearch
0.3.6
processing
2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nyalldawson**
Fixes https://github.com/qgis/QGIS/issues/59107

Also adds tests for this, as there were zero existing tests covering this functionality

### CODE REVIEW

File: src/core/dxf/qgsdxfexport.cpp [hunk 1]
```diff
@@ -2439,21 +2439,21 @@ void QgsDxfExport::drawLabel( const QString &layerId, QgsRenderContext &context,
   if ( tmpLyr.multilineAlign == Qgis::LabelMultiLineAlignment::FollowPlacement )
   {
     //calculate font alignment based on label quadrant
-    switch ( label->getQuadrant() )
+    switch ( label->quadrant() )
     {
-      case pal::LabelPosition::QuadrantAboveLeft:
-      case pal::LabelPosition::QuadrantLeft:
-      case pal::LabelPosition::QuadrantBelowLeft:
+      case Qgis::LabelQuadrantPosition::AboveLeft:
```
Comment (nirvn) [inline src/core/dxf/qgsdxfexport.cpp:2444]: Nice to see the pal:: gone in favor of Qgis:: here.

**Review by nirvn — APPROVED** [review]

**Comment by agiudiceandrea** [comment]
Please have a look at https://github.com/qgis/QGIS/issues/63155.
